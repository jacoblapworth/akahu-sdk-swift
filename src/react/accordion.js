import XUIAccordion from './components/accordion/XUIAccordion';

export default XUIAccordion;
