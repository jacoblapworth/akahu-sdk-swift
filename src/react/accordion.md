<div class="xui-margin-vertical">
		<a href="../section-compounds-displayingdata-accordion.html" isDocLink>Accordion in the XUI Documentation</a>
</div>

Accordions are used to display a vertically expandable & collapsible list that reveals and hides additional content

The _Accordion_ is broken up into two key components `import XUIAccordion, { XUIAccordionItem } from '@xero/xui/react/accordion';`.



## Basic

The bare minimum _Accordion_ composition can be achieved with the `items` and `createItem` prop.

+ `items` represents the generic data structure of each _Accordion_ item.
+ `createItem` returns a `<XUIAccordionItem />` that represents a single _item_ from the `items` prop.

**Note:** Each item needs a unique `id`. By default an object _key_ named `id` is looked up in each item however you can specify a new key if there is a more relevant unique identifier using the prop `idKey`.

```
const {
	default: XUIAccordion,
	XUIAccordionItem,
} = require('./accordion');
<XUIAccordion
	items={[
		{ id: 1, name: 'John Smith', content: 'Accountant' },
		{ id: 2, name: 'Barry Allen', content: 'Bookkeeper' },
		{ id: 3, name: 'Ernest Hemmingway' }
	]}
	createItem={({ name, content }) => (
		<XUIAccordionItem
			primaryHeading={name}
			onItemClick={({id, isOpen}) => console.log(`clicked! ${id} | ${isOpen}`)}>
			{ content &&  <div className="xui-padding-horizontal-large xui-padding-vertical-4xlarge">{content}</div> }
		</XUIAccordionItem>
	)}
/>
```

## Accordion Item

The `<XUIAccordionItem />` can take a variety of props that can inject content into the layout (`primaryHeading`, `secondaryHeading`, `leftContent`, `pinnedValue`, `action`, `overflow`, `custom`).

You can also supply an `onItemClick` prop that returns the entire item from the items array in addition to an `isOpen` boolean representing the items toggled state.

**Note:** if you are adding a `<button />` type element as content remember to [`stopPropagation`](https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation) so the event does not bubble up and trigger the _Accordion_ _expand_ / _collapse_ toggle accidentally.

```
const {
	default: XUIAccordion,
	XUIAccordionItem,
} = require('./accordion');
const itemStyle = { background: 'lightgray', border: '1px solid darkgray', padding: '4px', outline: '1px solid gray' };
<XUIAccordion
	items={[{ id: 1 }]}
	createItem={() => (
		<XUIAccordionItem
			primaryHeading={<div style={itemStyle}>Primary Heading</div>}
			secondaryHeading={<div style={itemStyle}>Secondary Heading</div>}
			leftContent={<div style={itemStyle}>Left Content</div>}
			pinnedValue={<div style={itemStyle}>Pinned Value</div>}
			action={<div style={itemStyle}>Action</div>}
			overflow={<div style={itemStyle}>Overflow</div>}
			custom={<div style={itemStyle}>Custom Content</div>}
			onItemClick={({isOpen}) => alert(`${isOpen ? 'Open' : 'Close'} example item`)}
		/>
	)}
/>
```

## Empty State

If no `children` content is supplied to a `<XUIAccordionItem />` then the _empty state_ component will be rendered instead.

There is a default design that can be overridden or completely replaced (`emptyMessage`, `emptyIcon`, `emptyStateComponent`).

```
const {
	default: XUIAccordion,
	XUIAccordionItem,
} = require('./accordion');
const starIconPath = require ('@xero/xui-icon/icons/star').default;
const XUIAvatar = require('./avatar').default;
const XUIButton = require('./button').default;
const XUIIcon = require('./icon').default;
const icon = require('@xero/xui-icon/icons/overflow').default;
<div>
	<XUIAccordion
		className="xui-margin-bottom-large"
		items={[{ id: 1 }]}
		createItem={({ name }) => <XUIAccordionItem primaryHeading="Default empty state" />}
	/>
	<XUIAccordion
		className="xui-margin-bottom-large"
		emptyIcon={starIconPath}
		emptyMessage="Custom empty state message"
		items={[{ id: 1 }]}
		createItem={({ name }) => <XUIAccordionItem primaryHeading="Custom empty state" />}
	/>
	<XUIAccordion
		emptyStateComponent={(
			<div
				className="xui-heading xui-textcolor-inverted xui-padding-vertical-large xui-text-align-center"
				style={ { background: 'darkgray' } }>
				Replace empty state component
			</div>
		)}
		items={[{ id: 1 }]}
		createItem={({ name }) => <XUIAccordionItem primaryHeading="Replace empty state" />}
	/>
</div>
```

## Example

You can nest other XUI component inside the `<XUIAccordionItem />` _(such as `<XUIContentBlock />`)_ to create robust compositions.

```
const {
	default: XUIAccordion,
	XUIAccordionItem,
} = require('./accordion');
const { XUIContentBlock, XUIContentBlockItem } = require('./structural.js');
const { default: XUIAvatar } = require('./avatar.js');
const { default: XUIButton } = require('./button.js');
const { default: XUIIcon } = require('./icon.js');
const { default: overflowPathData } = require('@xero/xui-icon/icons/overflow');
<XUIAccordion
	idKey="name"
	items={[
		{
			name: 'John Smith',
			contacts: [
				{ contact: 'Peggy Olsen', minutes: '00:20' },
				{ contact: 'Pete Campbell', minutes: '15:30' }
			]
		}
	]}
	createItem={({ name, contacts }) => (
		<XUIAccordionItem
			primaryHeading={name}
			leftContent={<XUIAvatar value={name} className="xui-margin-right"/>}
			overflow={<XUIButton variant="icon"><XUIIcon icon={overflowPathData} title="Overflow menu"/></XUIButton>}
			action={(
				<XUIButton
					size="small"
					className="xui-margin-right-small"
					onClick={event => {
						event.stopPropagation();
						alert('Clicked update button');
					}}>Update</XUIButton>
			)}>
				<XUIContentBlock>
					{ contacts.map(({contact, minutes}) => (
						<XUIContentBlockItem
							key={contact}
							primaryHeading={contact}
							pinnedValue={minutes}
							href="#"
						/>
					)) }
				</XUIContentBlock>
		</XUIAccordionItem>
	)}
/>
```
