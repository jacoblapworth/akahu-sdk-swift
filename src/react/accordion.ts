import XUIAccordion from './components/accordion/XUIAccordion';
import XUIAccordionItem from './components/accordion/XUIAccordionItem';

export { XUIAccordion as default, XUIAccordion, XUIAccordionItem };
