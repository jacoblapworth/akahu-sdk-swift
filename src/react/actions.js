import XUIActions from './components/actions/XUIActions';

export default XUIActions;
