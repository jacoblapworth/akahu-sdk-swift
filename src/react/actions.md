<div class="xui-margin-vertical">
	<a href="../section-components-controls-button.html#components-controls-button-12" isDocLink>Actions in the XUI Documentation</a>
</div>

Actions are used to wrap a selection of buttons (often two) and provide formatting. They can be embedded in a XUIModal, in a XUIPageHeader, in the footer of a XUIPanel, and other places. These are more generic than Banner Actions or Toast Actions.

#### Default action

A default action aligns a primary and secondary button to the right, with the primary as the rightmost.

```jsx harmony
import XUIActions from '@xero/xui/react/actions';
import XUIButton from '@xero/xui/react/button';

<XUIActions
  primaryAction={<XUIButton variant="main">Continue</XUIButton>}
  secondaryAction={<XUIButton>Cancel</XUIButton>}
/>;
```

#### Linear action

The linear variant aligns the primary button to the right and the secondary button to the left.

```jsx harmony
import XUIActions from '@xero/xui/react/actions';
import XUIButton from '@xero/xui/react/button';

<XUIActions
  isLinear
  primaryAction={<XUIButton variant="main">Next</XUIButton>}
  secondaryAction={<XUIButton>Previous</XUIButton>}
/>;
```
