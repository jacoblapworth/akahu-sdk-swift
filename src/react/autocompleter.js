import Autocompleter from './components/autocompleter/Autocompleter';
import AutocompleterInput from './components/autocompleter/AutocompleterInput';
import SecondarySearch from './components/autocompleter/SecondarySearch';
import EmptyState from './components/autocompleter/EmptyState';
import { boldMatch, decorateSubStr } from './components/autocompleter/helpers/highlighting';

export {
	Autocompleter as default,
	AutocompleterInput,
	EmptyState,
	SecondarySearch,
	boldMatch,
	decorateSubStr
};
