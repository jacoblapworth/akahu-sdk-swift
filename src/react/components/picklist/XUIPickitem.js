import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import PickitemBody from './private/PickitemBody';
import PickitemMultiselect from './private/PickitemMultiselect';
import { pickitemClassName, sideElementClassName } from './private/constants';
import { verticalOnlyProp } from './private/helpers';
import shouldRender from '../helpers/shouldRender';

/**
 * Presentational component used to display a selectable item in a
 * list of items.
 *
 * @export
 * @class XUIPickitem
 * @extends {PureComponent}
 */
export default class XUIPickitem extends PureComponent {
  render() {
    const pickItem = this;
    const {
      id,
      isSelected,
      className,
      onBlur,
      onFocus,
      onKeyDown,
      onClick,
      onMouseOver,
      onMouseUp,
      ariaRole,
      disableSelectedStyles,
      isMultiselect,
      href,
      children,
      qaHook,
      checkboxClassName,
      isHighlighted,
      isDisabled,
      isSplit,
      ariaLabel,
      target,
      shouldTruncate,
      pickitemBodyProps,
      leftElement,
      rightElement,
      primaryElement,
      secondaryElement,
      isInvalid,
      pinnedElement,
      headingElement,
      isMultiline,
      _isHorizontal,
    } = pickItem.props;

    const truncationClassName = (shouldTruncate && `${pickitemClassName}-text-truncated`) || '';

    let pickItemMinWidthClassName; // This is used to set min-width for pickItem when truncating
    if (shouldTruncate && !rightElement && !pinnedElement && (leftElement || secondaryElement)) {
      const leftElementClassName = cn(leftElement && '-leftelement');
      const secondaryElementClassName = cn(secondaryElement && '-secondarytext');
      pickItemMinWidthClassName = `${pickitemClassName}-has${leftElementClassName}${secondaryElementClassName}`;
    }
    const classes = cn(
      `${pickitemClassName}`,
      className,
      truncationClassName,
      isHighlighted && `${pickitemClassName}-is-hovered`,
      isMultiline && `${pickitemClassName}-multiline`,
      _isHorizontal && `${pickitemClassName}-is-horizontal`,
      isSelected && !disableSelectedStyles && `${pickitemClassName}-is-selected`,
      isMultiselect && `${pickitemClassName}-multiselect`,
      isSplit && `${pickitemClassName}-split`,
      isDisabled && `${pickitemClassName}-is-disabled`,
      `${pickitemClassName}-medium`,
      isInvalid && `${pickitemClassName}-is-invalid`,
      pickItemMinWidthClassName,
    );
    const listeners = !isDisabled
      ? {
          onClick,
          onBlur,
          onFocus,
          onKeyDown,
          onMouseOver,
          onMouseUp,
        }
      : null;

    const Tag = isSplit ? 'div' : 'li';
    const BodyComponent = isMultiselect ? PickitemMultiselect : PickitemBody;
    const itemRole = isSplit ? undefined : ariaRole;

    const wrappedLeft = leftElement && <span className={sideElementClassName}>{leftElement}</span>;
    const wrappedRight = rightElement && (
      <span className={sideElementClassName}>{rightElement}</span>
    );
    const secondaryWrapped = shouldRender(secondaryElement) && (
      <span className={cn(`${pickitemClassName}--secondary`, truncationClassName)}>
        {secondaryElement}
      </span>
    );
    const pinnedWrapped = shouldRender(pinnedElement) && (
      <span className={cn(`${pickitemClassName}--pinned`, truncationClassName)}>
        {pinnedElement}
      </span>
    );
    const headingWrapped = shouldRender(headingElement) && (
      <span className={cn(`${pickitemClassName}--heading`, truncationClassName)}>
        {headingElement}
      </span>
    );

    return (
      <Tag
        aria-label={ariaLabel}
        aria-selected={isMultiselect ? isSelected : null}
        className={classes}
        data-automationid={qaHook}
        id={id}
        role={itemRole}
      >
        <BodyComponent
          {...{
            shouldTruncate,
            isSelected,
            isDisabled,
            href,
            checkboxClassName,
            target,
            primaryElement,
            ...pickitemBodyProps,
            ...listeners,
          }}
          headingElement={headingWrapped}
          leftElement={wrappedLeft}
          pinnedElement={pinnedWrapped}
          qaHook={qaHook && `${qaHook}--body`}
          rightElement={wrappedRight}
          secondaryElement={secondaryWrapped}
        >
          {children}
        </BodyComponent>
      </Tag>
    );
  }
}

XUIPickitem.propTypes = {
  /**
   * Inherited. Whether parent list is set to render horizontal pickitems. Do not set directly.
   * @ignore
   */
  _isHorizontal: PropTypes.bool,
  /** Optional label to add to a pickitem */
  ariaLabel: PropTypes.string,
  /** ARIA attribute defining what purpose this item serves. */
  ariaRole: PropTypes.string,
  /**
   * Classes can be passed to the `XUICheckbox` component in `XUIPickitemBody`.<br>
   * ⚠️ *Vertical picklists only*
   */
  checkboxClassName(...parameters) {
    return verticalOnlyProp(PropTypes.string, ...parameters);
  },
  children: PropTypes.node,
  className: PropTypes.string,
  /** For nested children such as checkboxes, icons or groups selected styles should be disabled. */
  disableSelectedStyles: PropTypes.bool,
  /**
   * Text to appear bolded as the first line. Pushes secondary to a new line.<br>
   * ⚠️ *Vertical picklists only*
   */
  headingElement(...parameters) {
    return verticalOnlyProp(PropTypes.node, ...parameters);
  },
  /** Link to be used in child, will render an a tag if used and button if not */
  href: PropTypes.string,
  /** id must be unique and unchangeable  */
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  /** The disabled behaviour and styles are applied when this is true. */
  isDisabled: PropTypes.bool,
  /** Is the item highlighted? */
  isHighlighted: PropTypes.bool,
  /** Is the item invalid? */
  isInvalid: PropTypes.bool,
  /**
   * Allows flex-wrap, achieving a look much like content-block. Use for AutoCompleter.<br>
   * ⚠️ *Vertical picklists only*
   */
  isMultiline(...parameters) {
    return verticalOnlyProp(PropTypes.bool, ...parameters);
  },
  /**
   * When true a checkbox will be added to the layout of the child component.<br>
   * ⚠️ *Vertical picklists only*
   */
  isMultiselect(...parameters) {
    return verticalOnlyProp(PropTypes.bool, ...parameters);
  },
  /** Is the item selected? */
  isSelected: PropTypes.bool,
  /**
   * Whether or not this pickitem sits next to a `XUINestedPicklistToggle`.<br>
   * ⚠️ *Vertical picklists only*
   */
  isSplit(...parameters) {
    return verticalOnlyProp(PropTypes.bool, ...parameters);
  },
  /** Content to be added to the left of the pickitem. */
  leftElement: PropTypes.node,
  /** Callback on blur of the pickitem */
  onBlur: PropTypes.func,
  /** Callback when the pickitem is clicked. */
  onClick: PropTypes.func,
  /** Callback on focus of the pickitem */
  onFocus: PropTypes.func,
  /** Callback on keydown of the pickitem */
  onKeyDown: PropTypes.func,
  /** Callback on mouseup of the pickitem */
  onMouseUp: PropTypes.func,
  /** Callback when this item is selected by a parent component */
  onSelect: PropTypes.func,
  /** Props to pass to the pickitem body */
  pickitemBodyProps: PropTypes.object,
  /**
   * Less important text to appear pinned at the right. Can be plain text.<br>
   * ⚠️ *Vertical picklists only*
   */
  pinnedElement(...parameters) {
    return verticalOnlyProp(PropTypes.node, ...parameters);
  },
  /** Standard text. Can be plain text. */
  primaryElement: PropTypes.node,
  /** The automation-id to add to the item */
  qaHook: PropTypes.string,
  /**
   * Content to be added to the right of the pickitem.<br>
   * ⚠️ *Vertical picklists only*
   */
  rightElement(...parameters) {
    return verticalOnlyProp(PropTypes.node, ...parameters);
  },
  /** Less important text to appear beside primary. Can be plain text. */
  secondaryElement: PropTypes.node,
  /** Whether to truncate text instead of wrapping. Where possible, please set this on the
   * containing picklist, which will override any per-item settings. */
  shouldTruncate: PropTypes.bool,
  /** When a link is preferred, this target prop can be used on the <a> tag */
  target: PropTypes.string,
  /** The value associated with this PickItem which will be passed to the onSelect callbacks
   * here and in the stateful picklist */
  value: PropTypes.any,
};

XUIPickitem.defaultProps = {
  ariaRole: 'option',
  disableSelectedStyles: false,
  isDisabled: false,
  isSelected: false,
  isSplit: false,
  /*
	 DO NOT REMOVE
	 This property is needed so that the stateful picklist will properly recognize this
	 component as a menu item.
	 */
  _isMenuItem: true, // eslint-disable-line
};
