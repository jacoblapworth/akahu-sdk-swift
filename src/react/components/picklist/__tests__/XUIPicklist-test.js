import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { nanoid } from 'nanoid';
import { axe, toHaveNoViolations } from 'jest-axe';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { screen } from '@testing-library/dom';
import XUIPicklist from '../XUIPicklist';
import XUIPickitem from '../XUIPickitem';

Enzyme.configure({ adapter: new Adapter() });
expect.extend(toHaveNoViolations);

jest.mock('nanoid');
nanoid.mockImplementation(() => 'testPickitemCheckboxId');

const setup = (props = {}, itemProps = {}) => {
  const expected = renderer.create(
    <XUIPicklist {...props}>
      <XUIPickitem id="pi1" primaryElement="Item content" {...itemProps} />
    </XUIPicklist>,
  );

  return expected;
};

describe('< Picklist />', () => {
  it('renders most basic example', () => {
    const basic = setup();
    expect(basic).toMatchSnapshot();
  });

  it('should pass accessibility testing', async () => {
    const wrapper = mount(
      <XUIPicklist>
        <li>Pickitem</li>
      </XUIPicklist>,
    );
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it('example with most compatible options', () => {
    const settings = {
      className: 'custom-picklist-class',
      id: 'picklistId',
      isHorizontal: true,
      hasDefaultLayout: false,
      shouldTruncate: true,
      secondaryProps: { role: 'presentation' },
      qaHook: 'picklist-example',
    };
    const mostOptions = setup(settings);
    expect(mostOptions).toMatchSnapshot();
  });

  it('example of list settings overriding item-level settings', () => {
    const settings = {
      isMultiselect: false,
      shouldTruncate: true,
    };
    const itemProps = {
      isMultiselect: true,
    };
    const overrides = setup(settings, itemProps);
    expect(overrides).toMatchSnapshot();
  });

  it('example taking settings from first child and giving to all', () => {
    const fromItems = renderer.create(
      <XUIPicklist>
        <XUIPickitem id="pi1" isMultiselect primaryElement="Item content" />
        <XUIPickitem id="pi2" primaryElement="Item two" />
      </XUIPicklist>,
    );
    expect(fromItems).toMatchSnapshot();
  });

  it('should not override a manually set ariaRole on a child XUIPickitem', () => {
    render(
      <XUIPicklist secondaryProps={{ role: 'menu' }}>
        <XUIPickitem id="pi1" qaHook="pi1" ariaRole="menuitem" />
      </XUIPicklist>,
    );

    const pickItem = screen.getByRole('menuitem');
    expect(pickItem).toBeTruthy();
  });

  it("should default the ariaRole of a child XUIPickitem to 'option' when the XUIPicklist role is listbox", () => {
    render(
      <XUIPicklist secondaryProps={{ role: 'listbox' }}>
        <XUIPickitem id="pi1" qaHook="pi1" />
      </XUIPicklist>,
    );

    const pickItem = screen.getByRole('option');
    expect(pickItem).toBeTruthy();
  });
});
