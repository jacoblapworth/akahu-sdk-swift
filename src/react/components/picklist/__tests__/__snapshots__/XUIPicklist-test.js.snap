// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Picklist /> example of list settings overriding item-level settings 1`] = `
<ul
  className="xui-picklist xui-picklist-layout xui-picklist-medium"
  onChange={[Function]}
>
  <li
    aria-selected={null}
    className="xui-pickitem xui-pickitem-text-truncated xui-pickitem-medium"
    id="pi1"
  >
    <button
      className="xui-pickitem--body"
      rel={null}
      type="button"
    >
      <span
        className="xui-pickitem--text xui-pickitem-text-truncated"
      >
        <span
          className="xui-pickitem-text-truncated"
        >
          Item content
        </span>
      </span>
    </button>
  </li>
</ul>
`;

exports[`< Picklist /> example taking settings from first child and giving to all 1`] = `
<ul
  className="xui-picklist xui-picklist-layout xui-picklist-medium"
  onChange={[Function]}
>
  <li
    aria-selected={false}
    className="xui-pickitem xui-pickitem-multiselect xui-pickitem-medium"
    id="pi1"
  >
    <div
      className="xui-pickitem--body"
      role="button"
      tabIndex={0}
    >
      <div
        className="xui-pickitem-multiselect--checkbox xui-styledcheckboxradiowrapper xui-styledcheckboxradio-small"
      >
        <label
          className="xui-styledcheckboxradio"
          onClick={[Function]}
        >
          <input
            aria-labelledby="xui-testPickitemCheckboxId"
            checked={false}
            className="xui-styledcheckboxradio--input xui-styledcheckboxradio--input-small xui-pickitem--input"
            disabled={false}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            type="checkbox"
          />
          <div
            className="xui-styledcheckboxradio--checkbox xui-pickitem--input xui-styledcheckboxradio--checkbox-small"
            onAnimationEnd={[Function]}
          >
            <div
              className="xui-touchtarget"
            />
          </div>
          <span
            className="xui-styledcheckboxradio--label xui-styledcheckboxradio--label-small xui-pickitem--text xui-pickitem-multiselect--label"
            id="xui-testPickitemCheckboxId"
          >
            <span
              className=""
            >
              Item content
            </span>
          </span>
        </label>
      </div>
    </div>
  </li>
  <li
    aria-selected={false}
    className="xui-pickitem xui-pickitem-multiselect xui-pickitem-medium"
    id="pi2"
  >
    <div
      className="xui-pickitem--body"
      role="button"
      tabIndex={0}
    >
      <div
        className="xui-pickitem-multiselect--checkbox xui-styledcheckboxradiowrapper xui-styledcheckboxradio-small"
      >
        <label
          className="xui-styledcheckboxradio"
          onClick={[Function]}
        >
          <input
            aria-labelledby="xui-testPickitemCheckboxId"
            checked={false}
            className="xui-styledcheckboxradio--input xui-styledcheckboxradio--input-small xui-pickitem--input"
            disabled={false}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            type="checkbox"
          />
          <div
            className="xui-styledcheckboxradio--checkbox xui-pickitem--input xui-styledcheckboxradio--checkbox-small"
            onAnimationEnd={[Function]}
          >
            <div
              className="xui-touchtarget"
            />
          </div>
          <span
            className="xui-styledcheckboxradio--label xui-styledcheckboxradio--label-small xui-pickitem--text xui-pickitem-multiselect--label"
            id="xui-testPickitemCheckboxId"
          >
            <span
              className=""
            >
              Item two
            </span>
          </span>
        </label>
      </div>
    </div>
  </li>
</ul>
`;

exports[`< Picklist /> example with most compatible options 1`] = `
<ul
  className="xui-picklist xui-picklist-medium custom-picklist-class xui-picklist-horizontal"
  data-automationid="picklist-example"
  id="picklistId"
  onChange={[Function]}
  role="presentation"
>
  <li
    aria-selected={null}
    className="xui-pickitem xui-pickitem-text-truncated xui-pickitem-is-horizontal xui-pickitem-medium"
    id="pi1"
  >
    <button
      className="xui-pickitem--body"
      rel={null}
      type="button"
    >
      <span
        className="xui-pickitem--text xui-pickitem-text-truncated"
      >
        <span
          className="xui-pickitem-text-truncated"
        >
          Item content
        </span>
      </span>
    </button>
  </li>
</ul>
`;

exports[`< Picklist /> renders most basic example 1`] = `
<ul
  className="xui-picklist xui-picklist-layout xui-picklist-medium"
  onChange={[Function]}
>
  <li
    aria-selected={null}
    className="xui-pickitem xui-pickitem-medium"
    id="pi1"
  >
    <button
      className="xui-pickitem--body"
      rel={null}
      type="button"
    >
      <span
        className="xui-pickitem--text"
      >
        Item content
      </span>
    </button>
  </li>
</ul>
`;
